packages/astro/src/vite-plugin-build-css/index.ts

```diff
@@ -156,6 +176,34 @@ export function rollupPluginAstroBuildCSS(options: PluginOptions): VitePlugin {
 				.replace(/\./g, '\\.');
 			const emptyChunkRE = new RegExp(opts.format === 'es' ? `\\bimport\\s*"[^"]*(?:${emptyChunkFiles})";\n?` : `\\brequire\\(\\s*"[^"]*(?:${emptyChunkFiles})"\\);\n?`, 'g');
 
+			// Crawl the module graph to find CSS chunks to create
+			if(!legacy) {
+				for(const id of this.getModuleIds()) {
+					if(hasPageDataByViteID(internals, id)) {
+						let pageStyles = '';
+						for(const [_styleId, styles] of walkStyles(this, id)) {
+							pageStyles += styles;
+						}
+
+						// Pages with no styles, nothing more to do
+						if(!pageStyles) continue;
+
+						const { code: minifiedCSS } = await esbuild.transform(pageStyles, {
+							loader: 'css',
+							minify: true,
+						});
+						const referenceId = this.emitFile({
+							name: 'entry' + '.css',
+							type: 'asset',
+							source: minifiedCSS,
+						});
+						const fileName = this.getFileName(referenceId);
+
+						getPageDataByViteID(internals, id)?.css.add(fileName);
```

Comment:
worth an error if `getPageDataByViteID(internals, id)` doesn't appear? I can imagine this being error-worthy vs. silently failing

Reply:
No, we've already verified that this is a page's module with the `hasPageDataByViteID` check. Can add a clarifying comment.